src/v/serde/envelope_for_each_field.h

```diff
@@ -218,7 +213,7 @@ inline auto envelope_for_each_field(T& t, Fn&& fn) {
     }
 }
 
-template<is_envelope T, check_for_more_fn Fn>
+template<is_envelope T, std::predicate Fn>
```

Comment:
Sorry to be a pain, and go back on this, but something feels wrong. I don't think it's conceptually a predicate, but I'm not quite sure what it is. @felixguendling might be able to comment here.

Reply:
yeah, i like the original definition and name better. dropped the commit.